### ISSUE TITLE
Error occurs when only using partition columns in query

### ISSUE THREAD

**Issue by pjmore** [apache/arrow-datafusion#1999]
**Describe the bug**
When using a partitioned datasource selecting only the partitioned columns causes errors. 

**To Reproduce**
Create sample parquet/csv file content doesn't matter. Copy file to two separate paths using path partitioning scheme. E.g.
"year=2021/month=09/day=09/file.parquet",
"year=2021/month=10/day=09/file.parquet",

Register listing table and execute query 
```select distinct year,month, day from t```

**Expected behavior**
Should return record batch with: 
```
+------+-------+-----+
| year | month | day||
+------+-------+-----+
| 2021 | 09    | 09  |
| 2021 | 10    | 09  |
| 2021 | 10    | 28  |
+------+-------+-----+ 
```

**Additional context**
This was verified to be a problem with parquet and csv formats. Unsure about avro or json.

**Comment by Jefffrey** [comment]
Issue seems resolves as of 96512ac05df40deb42632992cece656842d6debb as unable to reproduce:

Created file structure as shown:

```
jeffrey:~/tmpdata$ pwd
/home/jeffrey/tmpdata
jeffrey:~/tmpdata$ tree *test
csvtest
└── year=2021
    ├── month=09
    │   └── day=09
    │       └── time.csv
    └── month=10
        └── day=09
            └── time.csv
parttest
└── year=2021
    ├── month=09
    │   └── day=09
    │       └── part-0.parquet
    └── month=10
        └── day=09
            └── part-0.parquet
```

Then selecting their partitioned columns:

```sql
DataFusion CLI v14.0.0
❯ create external table parttest stored as parquet partitioned by (year, month, day) location '/home/jeffrey/tmpdata/parttest';
0 rows in set. Query took 0.003 seconds.
❯ select distinct year, month, day from parttest;
+------+-------+-----+
| year | month | day |
+------+-------+-----+
| 2021 | 10    | 09  |
| 2021 | 09    | 09  |
+------+-------+-----+
2 rows in set. Query took 0.105 seconds.
❯ create external table csvtest stored as csv partitioned by (year, month, day) location '/home/jeffrey/tmpdata/csvtest';
0 rows in set. Query took 0.018 seconds.
❯ select distinct year, month, day from csvtest;
+------+-------+-----+
| year | month | day |
+------+-------+-----+
| 2021 | 09    | 09  |
| 2021 | 10    | 09  |
+------+-------+-----+
2 rows in set. Query took 0.008 seconds.
```

Can see selects fine with no errors.

**Comment by pjmore** [comment]
@Jefffrey Yeah you're right this was fixed by #2000 and I should've closed it then. Thanks for the heads up on this one.
